### DISCUSSION TITLE
Formula for counting child nodes in attributes needed

### DISCUSSION THREAD

**Discussion by paddeladdi** [freeplane/freeplane#1887] [Help]
Hi,
I was searching for hours and tried different formulas but didn´t find the right solution for my problem. 

I have a very large map. In all nodes I have an attribute with the value of the amount of direct child nodes for better overview. This value i put in manually, but as bigger as the map gets, counting of the child nodes gets more time-consuming. I´m sure there is a simple way with a formula to just count the amount of all child nodes of a node, but I cannot find the solution. 

Thanks for any help!
Addi

**Comment by macmarrum** [comment]
This formula considers all children, also technical (like for summary nodes) and hidden by a filter
```groovy
=node.children.size()
```
Here's a formula that ignores any invisible children, but this one is more tricky for Freeplane to evaluate and sometimes it presents stale values (when filters are applied)
```
=node.children.findAll { it.isVisible() }.size()
```

If you're interested in learning a bit about scripting in Freeplane, I recommend https://docs.freeplane.org/scripting/api-groovy-tutorial.html

**Reply by paddeladdi** [reply]
Thank you very much! The first formula is exactly what I was looking for and works perfectly for me!

I also read through the Groovy tutorial but I feel I´m a little too old to get this working in my brain ;)
